### PR TITLE
VEOSS-397 Payments integration

### PR DESCRIPTION
**Pull request by darbyo**

### CODE REVIEW

File: app/connectors/PaymentHttpParser.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+  implicit object ReturnPaymentReads extends HttpReads[ReturnPaymentResponse] {
+
+    override def read(method: String, url: String, response: HttpResponse): ReturnPaymentResponse = {
+      response.status match {
+        case CREATED =>
+          response.json.validate[PaymentResponse] match {
+            case JsSuccess(r, _) => Right(r)
+            case JsError(errors) =>
+              logger.warn(s"PaymentResponse failed trying to parse JSON $errors. Json was ${response.json}", errors)
```
Comment (TheNickWilson) [inline app/connectors/PaymentHttpParser.scala:38]: Is that a rogue $errors in the middle of this warning?  And should it be logged at warn, or error?

File: app/services/PaymentService.scala [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+package services
+
+import models.Period
+import models.requests.{PaymentPeriod, PaymentRequest}
+import uk.gov.hmrc.domain.Vrn
+
+import javax.inject.Inject
+
+class PaymentService @Inject()() {
```
Comment (TheNickWilson) [inline app/services/PaymentService.scala:25]: Given how simple this is (i.e. it doesn't need any other services etc. injected in), could it just be an apply method on the PaymentRequest companion object?

File: app/services/PaymentService.scala [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  def buildPaymentRequest(vrn: Vrn, period: Period, amount: Long): PaymentRequest = {
```
Comment (TheNickWilson) [inline app/services/PaymentService.scala:27]: Should amount here be a BigDecimal?  At this point I think it's the amount in pounds, so treating it as a long will round the value

**Review by TheNickWilson — CHANGES_REQUESTED** [review]

File: app/connectors/PaymentHttpParser.scala (hunk 1 above)
Comment (darbyo) [reply app/connectors/PaymentHttpParser.scala:38]: Will update to better logging, deffo should be error

File: app/services/PaymentService.scala (hunk 2 above)
Comment (darbyo) [reply app/services/PaymentService.scala:25]: Yep, will move it over and delete the service 👍🏻

**Review by TheNickWilson — APPROVED** [review]
Looks good, as discussed we can refactor how the pence amount is calculated (get it in one place, maybe look at using a Pence type instead of Long etc.) in another ticket 👍

**Pull request merged**